### PR TITLE
Migrate JAXRS WebContainer FAT testcases to Open Liberty

### PR DESCRIPTION
**Pull request by jim-krueger**

### CODE REVIEW

**Comment by jim-krueger** [comment]
#run-libby-bot

**Comment by jim-krueger** [comment]
#build

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/helloworld/HelloWorldTest.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+ * IBM Confidential
+ *
+ * OCO Source Materials
+ *
+ * Copyright IBM Corp. 2011
+ *
+ * The source code for this program is not published or other-
+ * wise divested of its trade secrets, irrespective of what has
+ * been deposited with the U.S. Copyright Office.
+ */
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/helloworld/HelloWorldTest.java:24]: I think this is what the copyright tool is complaining about.  Can you delete these comments?  The copyright header at the top looks right.

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/helloworld/HelloWorldTest.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+import java.io.File;
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/helloworld/HelloWorldTest.java:25]: This seems out of place.  Can you put the `import static` lines above the regular `import` lines?

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/helloworld/HelloWorldTest.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+
+@RunWith(FATRunner.class)
+public class HelloWorldTest {
+    @Server("com.ibm.ws.jaxrs2x.fat")
+    public static LibertyServer server;
+
+//    private final Class<?> c = HelloWorldTest.class;
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/helloworld/HelloWorldTest.java:53]:
```suggestion
```

Let's go ahead and remove any dead code.

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/multipart/MultipartTest.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+    @BeforeClass
+    public static void setup() throws Exception {
+
+        WebArchive app = ShrinkHelper.buildDefaultApp(thirdpartylibwar, "com.ibm.ws.jaxrs2x.fat.thirdparty.multipart");
+        ShrinkHelper.exportAppToServer(server, app);
+        server.addInstalledAppForValidation(thirdpartylibwar);
+
+        
+//        server.addInstalledAppForValidation(thirdpartylibwar);
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/multipart/MultipartTest.java:60]:
```suggestion
```

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/publish/servers/jaxrs2x.service.DisableJaxRS11/server.xml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<server>
+    <featureManager>
+        <!-- Cannot configure both at the same time <feature>jaxrs-1.1</feature> -->
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/publish/servers/jaxrs2x.service.DisableJaxRS11/server.xml:3]:
```suggestion
```

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/test-applications/thirdpartylib/src/com/ibm/ws/jaxrs2x/fat/thirdparty/multipart/UploadServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.MultivaluedMap;
+import javax.ws.rs.core.Response;
+
+import com.ibm.websphere.jaxrs20.multipart.IAttachment;
+
+//import org.apache.cxf.jaxrs.ext.multipart.Attachment;
```
Comment (andymc12) [inline dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/test-applications/thirdpartylib/src/com/ibm/ws/jaxrs2x/fat/thirdparty/multipart/UploadServiceImpl.java:34]:
```suggestion
```

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/helloworld/HelloWorldTest.java (hunk 1 above)
Comment (jim-krueger) [reply dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/helloworld/HelloWorldTest.java:24]: Done

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/helloworld/HelloWorldTest.java (hunk 2 above)
Comment (jim-krueger) [reply dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/helloworld/HelloWorldTest.java:25]: Done

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/helloworld/HelloWorldTest.java (hunk 3 above)
Comment (jim-krueger) [reply dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/helloworld/HelloWorldTest.java:53]: Done

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/multipart/MultipartTest.java (hunk 4 above)
Comment (jim-krueger) [reply dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/multipart/MultipartTest.java:60]: Done

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/publish/servers/jaxrs2x.service.DisableJaxRS11/server.xml (hunk 5 above)
Comment (jim-krueger) [reply dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/publish/servers/jaxrs2x.service.DisableJaxRS11/server.xml:3]: Done

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/test-applications/thirdpartylib/src/com/ibm/ws/jaxrs2x/fat/thirdparty/multipart/UploadServiceImpl.java (hunk 6 above)
Comment (jim-krueger) [reply dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/test-applications/thirdpartylib/src/com/ibm/ws/jaxrs2x/fat/thirdparty/multipart/UploadServiceImpl.java:34]: Done

**Comment by jim-krueger** [comment]
#run-libby-bot

**Comment by jim-krueger** [comment]
#run-libby-bot

**Comment by jim-krueger** [comment]
#build

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/build.gradle [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+dependencies {
+  jackson2x 'com.fasterxml.jackson.core:jackson-annotations:2.4.1',
+    'com.fasterxml.jackson.core:jackson-core:2.4.1',
+    'com.fasterxml.jackson.core:jackson-databind:2.4.1',
+    'com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:2.4.1',
+    'com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:2.4.1',
+    'com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.4.1'
+  requiredLibs 'commons-httpclient:commons-httpclient:3.1',
+    'commons-logging:commons-logging:1.2',
+	'org.apache.httpcomponents:httpclient:4.3.1',
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/build.gradle:44]: your spacing is uneven :)

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/disablejaxrs11/DisableJaxRS11Test.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.log.Log;
+
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.impl.LibertyServerFactory;
+
+/**
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/disablejaxrs11/DisableJaxRS11Test.java:26]: delete empty comment

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/jsonp/JaxRsJsonPTest.java [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+import com.ibm.websphere.simplicity.log.Log;
+import com.ibm.ws.jaxrs2x.fat.TestUtils;
+
+import componenttest.annotation.AllowedFFDC;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import componenttest.topology.impl.LibertyServerFactory;
+
+/**
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/fat/src/com/ibm/ws/jaxrs2x/fat/jsonp/JaxRsJsonPTest.java:34]: delete empty comment

File: dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/test-applications/jsonp/src/com/ibm/ws/jaxrs2x/fat/jsonp/service/TestApplication.java [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+ *******************************************************************************/
+
+package com.ibm.ws.jaxrs2x.fat.jsonp.service;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.ws.rs.core.Application;
+
+/**
```
Comment (WhiteCat22) [inline dev/com.ibm.ws.jaxrs.2.x.webcontainer_fat/test-applications/jsonp/src/com/ibm/ws/jaxrs2x/fat/jsonp/service/TestApplication.java:19]: delete empty comment

**Comment by jim-krueger** [comment]
#build

**Comment by jim-krueger** [comment]
#run-libby-bot

**Pull request merged**